wayback-core/src/main/java/org/archive/wayback/replay/charset/UniversalChardetSniffer.java

```diff
@@ -29,16 +29,24 @@ public String sniff(Resource resource) {
 
 			detector.setText(bbuffer);
 			CharsetMatch[] matches = detector.detectAll();
-			if (matches != null && matches.length > 0) {
-				charsetName = matches[0].getName();
+			for (int i = 0; matches != null && i < matches.length; i++) {
```

Comment:
Just for the sake of cleanliness, I suggest moving `matches != null` check out of the loop.